### ISSUE TITLE
Pass xdg-activation token through `alacritty msg create-window`

### ISSUE THREAD

**Issue by YaLTeR** [alacritty/alacritty#8337]
Currently, alacritty correctly activates itself on opening using the token from the compositor (`XDG_ACTIVATION_TOKEN`). However, spawning a new window through `alacritty msg create-window` doesn't thread through the token, so the new window cannot activate itself.

My use-case is a compositor bind to spawn a terminal which looks like: `alacritty msg create-window || alacritty`. This way, if an alacritty process already exists, it is re-used for (slightly) faster startup and less resource use. However, this currently doesn't work with xdg-activation because the token isn't passed through `msg create-window`.

I imagine it can become more of a problem as compositors start to adopt stricter new window autofocus policies, or it can break the various "show spinning cursor until the app opens" things.

### System

OS: Linux
Version: alacritty 0.14.0
Linux/BSD: Wayland, niri

**Comment by chrisduerr** [comment]
Honestly if it was up to me, it probably wouldn't work at all. The only reason why it does is because Winit just handles it automatically.

The idea of capturing a random string and sending it along the IPC message doesn't really excite me. Though I guess it would essentially be no effort to do so?

Generally speaking though I'd consider any compositor behaving strangely without it just bad, so I'm not particularly worried about it.

**Comment by kchibisov** [comment]
I guess the point is to avoid keyboard focus stealing like you can do with `wl-copy` where you don't have the `wlr_data_control` around, so windows don't get randomly focused to steal the keyboard input.

Generally passing token is fairly simple.

**Comment by chrisduerr** [comment]
> I guess the point is to avoid keyboard focus stealing like you can do with wl-copy where you don't have the wlr_data_control around, so windows don't get randomly focused to steal the keyboard input.

Yeah but people actually **want** that though, right? Also don't we unset the activation token after startup anyway, so it's never visible to applications running inside of Alacritty?

**Comment by kchibisov** [comment]
yeah, we unset it, since it shouldn't be passed further. The by design is _not focus steal_ and i think the token is the only thing that kind of makes it work.

It does mean that if you run e.g. `wl-copy` via something that activates it'll work, but it means that _user_ wanted to run it in a first place.
